cpp/test/DataStorm/config/Writer.cpp

```diff
@@ -308,10 +308,18 @@ void ::Writer::run(int argc, char* argv[])
     }
     cout << "ok" << endl;
 
+    struct WriterHolder
+    {
+        Ice::CommunicatorHolder communicator;
+        Node node;
+        Topic<string, int> topic;
+        SingleKeyWriter<string, int> writer;
+    };
+
     cout << "testing send time discard policy... " << flush;
     {
         writers.update(false); // Not ready
-        vector<tuple<Ice::CommunicatorHolder, Node, Topic<string, int>, SingleKeyWriter<string, int>>> w;
+        vector<WriterHolder> w;
```

Comment:
This is to avoid the undefined destruction order of tuples, we don't want the communicator to be destroyed before than the node.